apps/admin_app/lib/admin_app_web/views/product_view.ex

```diff
@@ -16,6 +24,11 @@ defmodule AdminAppWeb.ProductView do
   @sort_field_keys ["rummage", "sort", "field"]
   @sort_order_keys ["rummage", "sort", "order"]
 
+  def active_stock_locaton() do
+    StockLocation.active()
+    |> Enum.map(fn x -> {x.name, x.id} end)
+  end
+
   def themes_options(product) do
```

Comment:
@SagarKarwande are you sure about placing this method in the `product_view`? can you justify the responsibility of the file and the this method?

Reply:
We write some function in `view` file that are specifically needed for templates.

`active_stock_locaton/0` is used to show select dropdown for active Stock Locations. Phoenix forms `select` need the data in one of the following forms:
```elixir
[[key: "Pune", value: 1 ], [key: "Mumbai", value: 2]] OR
["Pune": 1, "Mumbai": 2] OR
[{"Pune", 1}, {"Mumbai", 2}]
```
This is something that is specific that is needed by Phoenix. So I have used the `StockLocation` model to fetch the data and later transform it in to the way the `select` input needs it.

[Select input doc](https://hexdocs.pm/phoenix_html/Phoenix.HTML.Form.html#select/4)
> I'll change the naming of variables in the method.

Reply:
We write some function in `view` file that are specifically needed for templates.

`themes_options/1` is used to show select dropdown for Variation themes. Phoenix forms `select` need the data in one of the following forms:
```elixir
[[key: "Color", value: 1 ], [key: "Color-size", value: 2]] OR
["Color": 1, "Color-Size": 2] OR
[{"Color", 1}, {"Color-Size", 2}]
```
This is something that is specific that is needed by Phoenix. So I have used the preloaded variation themes to fetch the data and later transform it into the way the `select` input needs it.

[Select input doc](https://hexdocs.pm/phoenix_html/Phoenix.HTML.Form.html#select/4)